homeassistant/components/knx/fan.py

```diff
@@ -66,10 +66,8 @@ def __init__(self, xknx: XKNX, config: ConfigType) -> None:
         # FanSpeedMode.STEP if max_step is set
         self._step_range: tuple[int, int] | None = (1, max_step) if max_step else None
 
-        self._attr_supported_features = (
-            SUPPORT_SET_SPEED | SUPPORT_OSCILLATE
-            if self._device.supports_oscillation
-            else SUPPORT_SET_SPEED
+        self._attr_supported_features = SUPPORT_SET_SPEED | (
+            self._device.supports_oscillation * SUPPORT_OSCILLATE
```

Comment:
Not sure which style to pick. 
- `| (bool * int)` is briefly and is a good fit to `|` binary operation imho.
- `| (int if bool else 0)` is probably easier to read but more verbose.
- extra if checks would add a lot of lines but is easy to read. But one would have to keep context when reading
```
self._attr_supported_features = int
if bool:
    self._attr_supported_features |= int
if another_bool:
    self._attr_supported_features |= another_int
```

Any preferences?

Reply:
Just keep it simple:

```py
self._attr_supported_features = SUPPORT_SET_SPEED
if self._device.supports_oscillation:
    self._attr_supported_features |= SUPPORT_OSCILLATE
```

3 simple lines, that everybody can ready and understand in a blink of an eye.

Reply:
Sure. It was maybe a poor example. In cover 3 `if` are required equaling 10 lines to define 1 attribute 4 times 🙃

Reply:
That is not really a concern performance-wise. If more complex things are needed, one can always decide to use the entity property method instead of the entity class attribute.